bin/catkin_prepare_release

```
@@ -332,14 +334,17 @@ def main():
         print(fmt("@{gf}Creating tag '@{boldon}%s@{boldoff}'..." % (tag_name)))
         tag_repository(base_path, vcs_type, tag_name, args.tag_prefix != '')
 
-        # confirm commands to push to remote repository
-        commands = push_changes(base_path, vcs_type, dry_run=True)
-        print(fmt('@{gf}The following commands will be executed to push the changes and tag to the remote repository:'))
-        for cmd in commands:
-            print(fmt('  @{bf}@{boldon}%s@{boldoff}' % ' '.join(cmd)))
-        if not args.non_interactive and not prompt_continue('Push the local commits and tags to the remote repository', default=True):
-            raise RuntimeError(fmt('@{rf}Skipping push, to finish the release execute the commands manually.'))
-        push_changes(base_path, vcs_type)
+        if not args.no_push:
+            # confirm commands to push to remote repository
+            commands = push_changes(base_path, vcs_type, dry_run=True)
+            print(fmt('@{gf}The following commands will be executed to push the changes and tag to the remote repository:'))
+            for cmd in commands:
+                print(fmt('  @{bf}@{boldon}%s@{boldoff}' % ' '.join(cmd)))
+            if not args.non_interactive and not prompt_continue('Push the local commits and tags to the remote repository', default=True):
+                raise RuntimeError(fmt('@{rf}Skipping push, to finish the release execute the commands manually.'))
+            push_changes(base_path, vcs_type)
+        else:
+            print(fmt('@{gf}Not pushing because --no-push was given'))
```

Well if the current branch is not tracking any remote, which happens to be my use case, then I don't see what commands to generate. What I could do is try to run `push_changes(..., dry_run=True)`, silently catch the exception if there is one and display the commands otherwise.